src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/ProductType.cs

```diff
@@ -2,49 +2,187 @@
+                    var semanticUrl = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.SemanticUrl;
+                    return semanticUrl;
+                });
+
+            Field<StringGraphType>(
+                "metaDescription",
+                description: "Get metaDescription for product.",
+                resolve: context =>
+                {
+                    var metaDescription = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.MetaDescription;
+                    return metaDescription;
+                });
+
+            Field<StringGraphType>(
+                "metaKeywords",
+                description: "Get metaKeywords for product.",
+                resolve: context =>
+                {
+                    var metaKeywords = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.MetaKeywords;
+                    return metaKeywords;
+                });
+
+            Field<StringGraphType>(
+                "metaTitle",
+                description: "Get metaTitle for product.",
+                resolve: context =>
+                {
+                    var metaTitle = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.PageTitle;
+                    return metaTitle;
+                });
+
+            Field<StringGraphType>(
+                "imgSrc",
+                description: "The product main image URL.",
+                resolve: context => context.Source.CatalogProduct.ImgSrc);
+
             Field(d => d.CatalogProduct.OuterId, nullable: true).Description("The outer identifier");
-            Field<ListGraphType<PropertyType>>("properties", resolve: context => context.Source.CatalogProduct.Properties);
 
-            Field<ListGraphType<PriceType>>("prices", arguments: new QueryArguments(new QueryArgument<StringGraphType> { Name = "currency", Description = "currency" }), resolve: context =>
+            Field<StringGraphType>(
+                "brandName",
+                description: "Get brandName for product.",
+                resolve: context =>
+                {
+                    var brandName = context.Source.CatalogProduct.Properties
+                        ?.FirstOrDefault(x => x.Name == "Brand")
+                        ?.Values
+                        ?.FirstOrDefault(x => x.Value != null)
+                        ?.Value;
+
+                    return brandName;
+                });
+
+            FieldAsync<VariationType>("masterVariation", resolve: async context =>
             {
-                var result = (context.Source).Prices;
-                var currency = context.GetArgument<string>("currency");
-                if (currency != null)
+                if (context.Source.CatalogProduct.MainProductId == null)
                 {
-                    result = result.Where(x => x.Currency.EqualsInvariant(currency)).ToList();
+                    var product = context.Source.CatalogProduct;
+
+                    return new ExpVariation
```

Comment:
Could you derive  `ExpVariation` from `ExpProduct` type and add ctor that receives the ExpProduct for typecasting? Then you will be able use such syntax `new ExpVariation(context.Source.CatalogProduct)`